src/components/organisms/JoinVenueModal/JoinVenueModal.tsx

```diff
@@ -0,0 +1,52 @@
+import React, { FC, Fragment, useCallback } from "react";
+import { Modal } from "react-bootstrap";
+
+import { AnyVenue } from "types/venues";
+
+import { joinVenue } from "utils/url";
+
+import "./JoinVenueModal.scss";
+
+interface JoinVenueModalProps {
+  venueId?: string;
+  venue: AnyVenue;
+  show: boolean;
+  isLoggedIn: boolean;
+  onHide?: () => void;
+}
+
+export const JoinVenueModal: FC<JoinVenueModalProps> = ({
+  show,
+  venueId,
+  venue,
+  isLoggedIn,
+  onHide,
+}) => {
+  const handleJoinVenue = useCallback(() => {
+    joinVenue(venueId, venue.entrance);
+  }, [venue.entrance, venueId]);
```

Comment:
Similar code is used in `src/components/molecules/VenueJoinButton/VenueJoinButton.tsx`

Under what cases are the modal used vs the button? Can the modal use the button?

Reply:
Modal is used to show text and redirect button after you enter the password or force you to login if you are not logged in and try to enter the venue.

The button is used to check the password and show the modal.

Reply:
Makes sense. I guess the part I was (am?) confused apart is why both of them have the `handleJoinVenue` code, and whether there is a simple/quick way that we avoid that duplication.

Reply:
I think if I prop drill it, I can have only 1 handleJoinVenue function.